### DISCUSSION TITLE
Is there a way to limit the resize height/width via URL?

### DISCUSSION THREAD

**Discussion by Spinks90** [SixLabors/ImageSharp.Web#324] [Q&A] [answered]
Is there a way to limit the width/height of an image when using the URL processor?

As it stands, my website would be vulnerable to a Denial of Service (DoS) attack, as setting a large value for height/width consumes all my memory on the server using URLs such as:
/images/test.jpg?width=100000&height=100000

Thanks.

**Comment by tocsoft** [comment, marked as answer]
there are a couple of way to handle this.
1. create a custom command parser (see [QueryCollectionRequestParser.cs](https://github.com/SixLabors/ImageSharp.Web/blob/main/src/ImageSharp.Web/Commands/QueryCollectionRequestParser.cs) for our default parser to copy) and limit your values there.
2. make use of our securing comands pipeline ([Securing commands docs](https://docs.sixlabors.com/articles/imagesharp.web/processingcommands.html#securing-processing-commands)) to lock down requests so they can't be tampered with to generate image sizes you don't want to allow.

**Reply by Spinks90** [reply]
> there are a couple of way to handle this.
> 
>     1. create a custom command parser (see [QueryCollectionRequestParser.cs](https://github.com/SixLabors/ImageSharp.Web/blob/main/src/ImageSharp.Web/Commands/QueryCollectionRequestParser.cs) for our default parser to copy) and limit your values there.
> 
>     2. make use of our securing comands pipeline ([Securing commands docs](https://docs.sixlabors.com/articles/imagesharp.web/processingcommands.html#securing-processing-commands)) to lock down requests so they can't be tampered with to generate image sizes you don't want to allow.

Ah, didn't spot that. Thank you!
